inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditReporterImpl.java

```diff
@@ -198,50 +219,201 @@ public void add(int auditID, String auditTag, String inlongGroupID, String inlon
         resetStat();
-        // Retrieve statistics from the list of objects without statistics to be eliminated
-        for (Map.Entry<String, StatInfo> entry : this.deleteCountMap.entrySet()) {
-            this.sumThreadGroup(entry.getKey(), entry.getValue());
+        LOGGER.info("pre stat map size {} {} {} {}", this.preStatMap.size(), this.expiredStatMap.size(),
+                this.summaryStatMap.size(), this.expiredKeyList.size());
+
+        summaryExpiredStatMap(isolateKey);
+
+        Iterator<Map.Entry<Long, ConcurrentHashMap<String, StatInfo>>> iterator = this.preStatMap.entrySet().iterator();
+        while (iterator.hasNext()) {
+            Map.Entry<Long, ConcurrentHashMap<String, StatInfo>> entry = iterator.next();
+            if (entry.getValue().isEmpty()) {
+                LOGGER.info("Remove the key of pre stat map: {},isolate key: {} ", entry.getKey(), isolateKey);
+                iterator.remove();
+                continue;
+            }
+            if (entry.getKey() > isolateKey) {
+                continue;
+            }
+            summaryPreStatMap(entry.getKey(), entry.getValue());
+            send(entry.getKey());
+
+        }
+
+        clearExpiredKey(isolateKey);
+
+        LOGGER.info("Finish report audit data");
+    }
+
+    /**
+     * Send base command
+     */
+    private void sendByBaseCommand(AuditApi.AuditRequest auditRequest) {
+        AuditApi.BaseCommand.Builder baseCommand = AuditApi.BaseCommand.newBuilder();
+        baseCommand.setType(AUDIT_REQUEST).setAuditRequest(auditRequest).build();
+        manager.send(baseCommand.build(), auditRequest);
+    }
+
+    /**
+     * Summary
+     */
+    private void sumThreadGroup(long isolateKey, String key, StatInfo statInfo) {
+        long count = statInfo.count.getAndSet(0);
+        if (0 == count) {
+            return;
+        }
+        ConcurrentHashMap<String, StatInfo> sumMap =
+                this.summaryStatMap.computeIfAbsent(isolateKey, k -> new ConcurrentHashMap<>());
+        StatInfo stat = sumMap.computeIfAbsent(key, k -> new StatInfo(0L, 0L, 0L));
+        stat.count.addAndGet(count);
+        stat.size.addAndGet(statInfo.size.getAndSet(0));
+        stat.delay.addAndGet(statInfo.delay.getAndSet(0));
+    }
+
+    /**
+     * Reset statistics
+     */
+    private void resetStat() {
+        dataId = 0;
+        packageId = 1;
+    }
+
+    /**
+     * Summary expired stat map
+     */
+    private void summaryExpiredStatMap(long isolateKey) {
+        Iterator<Map.Entry<Long, ConcurrentHashMap<String, StatInfo>>> iterator =
+                this.expiredStatMap.entrySet().iterator();
+        while (iterator.hasNext()) {
+            Map.Entry<Long, ConcurrentHashMap<String, StatInfo>> entry = iterator.next();
+            if (entry.getValue().isEmpty()) {
+                LOGGER.info("Remove the key of expired stat map: {},isolate key: {} ", entry.getKey(), isolateKey);
+                iterator.remove();
```

Comment:
It has synchronization issues between "entry.getValue().isEmpty()" and "iterator.remove()".